fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -2087,14 +2129,16 @@ private Money calculateSpecificDueDateChargeWithPercentage(final Money principal
     }
 
     private Money calculateInstallmentCharge(final PrincipalInterest principalInterestForThisPeriod, Money cumulative,
-            final LoanCharge loanCharge) {
+            final LoanCharge loanCharge, BigDecimal unutilizedAmount) {
         if (loanCharge.getChargeCalculation().isPercentageBased()) {
             BigDecimal amount = BigDecimal.ZERO;
             if (loanCharge.getChargeCalculation().isPercentageOfAmountAndInterest()) {
                 amount = amount.add(principalInterestForThisPeriod.principal().getAmount())
                         .add(principalInterestForThisPeriod.interest().getAmount());
             } else if (loanCharge.getChargeCalculation().isPercentageOfInterest()) {
                 amount = amount.add(principalInterestForThisPeriod.interest().getAmount());
+            } else if (loanCharge.getChargeCalculation().isPercentageOfUnutilizedAmount() && unutilizedAmount != null) {
+                amount = unutilizedAmount;
```

Comment:
Why are you passing the calculated unutilized amount and not just call `calcUnutilizedAmount` here?

Reply:
can be done it will require the loan terms as an argument instead of `unutilizedAmount`